### PR TITLE
Fix screen reader issues in calendar navigation buttons and combobox

### PR DESCRIPTION
**Pull request by conniec218**
## Current Behavior

[When activating the previous/next month/year buttons in the calendar component]
1. JAWS will not read the updated aria label for the current button, which means users will have to navigate away from the button then navigate back in order to know that the calendar date range has changed.
2. Both JAWS and Narrator will read the updated aria label of the header button before reading the updated aria label of the current button, but the header button should be read after/when the user is idle.

[Combobox]
1. JAWS does not read currently visible option when using the arrow keys to cycle through combobox options, but it should.

## New Behavior

[When activating the previous/next month/year buttons in the calendar component]
1. I added aria labels to each of the navigation buttons that match the button's title. JAWS will now read the button's updated aria label. This applies to all date range variations of the calendar.
2. I added a key to the CalendarMonth and CalendarYear header button, so that both JAWS and Narrator will read the header button aria label after updates to the current navigation button have been read (aka "politely").

[Combobox]
1. If no other title is supplied, I updated the title to be set to the combobox's current visible value. This way, JAWS will read the currently visible option when using the arrow keys to cycle through the available options.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/9b558cf266817f6499fece14e71bd90e98385a6b'>9b558cf266817f6499fece14e71bd90e98385a6b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/BaseButton-mount_base.html">929</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/BaseButton-mount.html">981</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Breadcrumb-mount_base.html">2828</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Breadcrumb-mount.html">2811</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Checkbox-mount_base.html">1574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Checkbox-mount.html">1536</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/CheckboxBase-mount_base.html">1324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/CheckboxBase-mount.html">1292</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ChoiceGroup-mount_base.html">4857</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ChoiceGroup-mount.html">4822</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ComboBox-mount_base.html">1046</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ComboBox-mount.html">1038</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/CommandBar-mount_base.html">10940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/CommandBar-mount.html">10840</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ContextualMenu-mount_base.html">11748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ContextualMenu-mount.html">11733</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DefaultButton-mount_base.html">1187</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DefaultButton-mount.html">1160</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRow-mount_base.html">3987</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRow-mount.html">3889</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRowFast-mount_base.html">3967</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRowFast-mount.html">4042</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRowNoStyles-mount_base.html">3772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DetailsRowNoStyles-mount.html">3752</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Dialog-mount_base.html">2257</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Dialog-mount.html">2304</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DocumentCardTitle-mount_base.html">186</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/DocumentCardTitle-mount.html">180</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Dropdown-mount_base.html">3355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Dropdown-mount.html">3414</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/FocusTrapZone-mount_base.html">1830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/FocusTrapZone-mount.html">1785</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/FocusZone-mount_base.html">1882</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/FocusZone-mount.html">1909</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/IconButton-mount_base.html">1868</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/IconButton-mount.html">1798</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Label-mount_base.html">348</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Label-mount.html">348</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Layer-mount_base.html">3140</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Layer-mount.html">3073</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Link-mount_base.html">496</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Link-mount.html">484</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/MenuButton-mount_base.html">1490</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/MenuButton-mount.html">1580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/MessageBar-mount_base.html">2221</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/MessageBar-mount.html">2176</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Nav-mount_base.html">3416</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Nav-mount.html">3399</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/OverflowSet-mount_base.html">1093</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/OverflowSet-mount.html">1103</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Panel-mount_base.html">2197</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Panel-mount.html">2336</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Persona-mount_base.html">998</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Persona-mount.html">1049</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Pivot-mount_base.html">1489</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Pivot-mount.html">1449</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/PrimaryButton-mount_base.html">1356</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/PrimaryButton-mount.html">1314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Rating-mount_base.html">7995</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Rating-mount.html">7948</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SearchBox-mount_base.html">1379</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SearchBox-mount.html">1340</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Shimmer-mount_base.html">2610</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Shimmer-mount.html">2628</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Slider-mount_base.html">1944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Slider-mount.html">1941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SpinButton-mount_base.html">5175</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SpinButton-mount.html">5233</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Spinner-mount_base.html">453</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Spinner-mount.html">454</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SplitButton-mount_base.html">3284</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SplitButton-mount.html">3293</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Stack-mount_base.html">523</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Stack-mount.html">526</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/StackWithIntrinsicChildren-mount_base.html">2377</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/StackWithIntrinsicChildren-mount.html">2392</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/StackWithTextChildren-mount_base.html">5476</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/StackWithTextChildren-mount.html">5358</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SwatchColorPicker-mount_base.html">11980</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/SwatchColorPicker-mount.html">12096</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TagPicker-mount_base.html">2978</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TagPicker-mount.html">2792</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TeachingBubble-mount_base.html">93716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TeachingBubble-mount.html">93971</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Text-mount_base.html">458</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Text-mount.html">436</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TextField-mount_base.html">1437</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/TextField-mount.html">1399</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-mount_base.html">1255</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-mount.html">1233</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-virtual-rerender_base.html">668</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-virtual-rerender.html">668</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1954</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1933</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Toggle-mount_base.html">809</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/Toggle-mount.html">847</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/buttonNative-mount_base.html">121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22613/perf-test/buttonNative-mount.html">128</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Calendar/CalendarMonth/CalendarMonth.base.tsx [hunk 1]
```diff
@@ -218,6 +225,7 @@ export const CalendarMonthBase: React.FunctionComponent<ICalendarMonthProps> = p
           aria-atomic={true}
           // if this component rerenders when text changes, aria-live will not be announced, so make key consistent
           aria-live="polite"
+          key={yearString}
```
Comment (jhellesen) [inline packages/react/src/components/Calendar/CalendarMonth/CalendarMonth.base.tsx:228]: Is the key change related to the JAWS behavior somehow?

Comment (conniec218) [reply packages/react/src/components/Calendar/CalendarMonth/CalendarMonth.base.tsx:228]: @jhellesen Yes; I'm not entirely sure why, but without the key, if we used the arrow navigation buttons, JAWS would read the header button's updated aria label (the button/div that shows what month/year/etc. it is) before the navigation button's aria label. By adding the key, JAWS now reads it after the navigation button and when the user isn't doing anything else, aka "politely". The key was already present in the CalendarDay component, so I'm adding it to CalendarMonth and CalendarYear here. 

I'll update the PR description to be a little more clear about this!

**Review by ermercer — APPROVED** [review]

**Review by jhellesen — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]

**Comment by smhigley** [comment]
Hey, sorry, I need to revert this since it creates some concerning behavior and duplicated announcements in other screen readers. I think one way forward would be to troubleshoot some of the live regions, since that should mitigate JAWS not announcing some of the Calendar changes. The Combobox issue should definitely be filed as a bug on JAWS as well.

**Comment by jhellesen** [comment]
> Hey, sorry, I need to revert this since it creates some concerning behavior and duplicated announcements in other screen readers. I think one way forward would be to troubleshoot some of the live regions, since that should mitigate JAWS not announcing some of the Calendar changes. The Combobox issue should definitely be filed as a bug on JAWS as well.

@smhigley could you expand what issues this caused and which screen readers were impacted?

**Comment by ermercer** [comment]
Adding @***@***.***> and @***@***.***> for awareness and to drive here
________________________________

**Pull request merged**